dcos/nodeutil/util.go

```diff
@@ -390,27 +390,38 @@ func (d *dcosInfo) state(ctx context.Context) (state State, err error) {
 	return state, err
 }
 
-// TaskCanonicalID return a CanonicalTaskID for a given task.
-func (d *dcosInfo) TaskCanonicalID(ctx context.Context, task string, completed bool) (*CanonicalTaskID, error) {
-	state, err := d.state(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	var foundTasks []Task
-	for _, framework := range state.Frameworks {
+func findTask(name string, completed bool, frameworks []Framework) (foundTasks []Task) {
+	for _, framework := range frameworks {
 		currentTasks := framework.Tasks
 		if completed {
 			currentTasks = framework.CompletedTasks
 		}
 
 		for _, t := range currentTasks {
-			if t.Name != task && !strings.Contains(t.ID, task) {
+			if t.Name != name && !strings.Contains(t.ID, name) {
 				continue
 			}
 			foundTasks = append(foundTasks, t)
 		}
 	}
+	return
+}
+
+// TaskCanonicalID return a CanonicalTaskID for a given task.
+func (d *dcosInfo) TaskCanonicalID(ctx context.Context, task string, completed bool) (*CanonicalTaskID, error) {
+	state, err := d.state(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	frameworksTasks := findTask(task, completed, state.Frameworks)
```

Comment:
It would be more readable to me to do something like: 

```
foundTasks := findTask(task, completed, state.Frameworks)
if completed {
  foundTasks = append(foundTasks, findTask(task, completed, state.CompletedFrameworks))
}
```